src/inference/trace_translators.jl

```diff
@@ -813,26 +864,32 @@ end
 
 Constructor for a symmetric trace translator.
 
-The involution is either constructed via the [`@transform`](@ref) macro (recommended), or can be provided as a Julia function.
+The involution is either constructed via the [`@transform`](@ref) macro (recommended),
+or can be provided as a Julia function.
 
 Run the translator with:
 
     (output_trace, log_weight) = translator(input_trace; check=false, observations=EmptyChoiceMap())
 
-Use `check` to enable the involution check (this requires that the transform `f` has been marked with [`is_involution`](@ref)).
+Use `check` to enable the involution check (this requires that the transform `f` has been
+marked with [`is_involution`](@ref)).
 
-If `check` is enabled, then `observations` is a choice map containing the observed random choices, and the check will additionally ensure they are not mutated by the involution.
+If `check` is enabled, then `observations` is a choice map containing the observed random
+choices, and the check will additionally ensure they are not mutated by the involution.
 """
-@with_kw struct SymmetricTraceTranslator{T <: Union{TraceTransformDSLProgram,Function}}
+@with_kw mutable struct SymmetricTraceTranslator{T <: TransformFunction} <: TraceTranslator
     q::GenerativeFunction
     q_args::Tuple = ()
     involution::T # an involution
 end
 
-function symmetric_trace_translator_run_transform(
-        involution::TraceTransformDSLProgram,
-        prev_model_trace::Trace, forward_proposal_trace::Trace,
-        q::GenerativeFunction, q_args::Tuple)
+function inverse(translator::SymmetricTraceTranslator, prev_model_trace=nothing)
```

Comment:
Format nitpick (if formatting changes are in scope :) ); that's a good candidate for:

```
inverse(translator::SymmetricTraceTranslator, prev_model_trace=nothing) = translator
```

(see first part of https://github.com/invenia/BlueStyle#method-definitions)